registration/tests/models.py

```diff
@@ -471,17 +474,332 @@ def test_resend_activation_email_expired_user(self):
+                                      'ACCOUNT_ACTIVATION_DAYS', None)
+        self.old_reg_email = getattr(settings,
+                                     'REGISTRATION_DEFAULT_FROM_EMAIL', None)
+        self.old_email_html = getattr(settings,
+                                      'REGISTRATION_EMAIL_HTML', None)
+        self.old_django_email = getattr(settings,
+                                        'DEFAULT_FROM_EMAIL', None)
+        self.old_admins = getattr(settings,
+                                  'ADMINS', None)
+
+        settings.ACCOUNT_ACTIVATION_DAYS = 7
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = 'registration@email.com'
+        settings.REGISTRATION_EMAIL_HTML = True
+        settings.DEFAULT_FROM_EMAIL = 'django@email.com'
+        settings.ADMINS = (
+            ('T-Rex', 'admin1@iamtrex.com'),
+            ('Flea', 'admin2@iamaflea.com')
+        )
+
+    def tearDown(self):
+        settings.ACCOUNT_ACTIVATION_DAYS = self.old_activation
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = self.old_reg_email
+        settings.REGISTRATION_EMAIL_HTML = self.old_email_html
+        settings.DEFAULT_FROM_EMAIL = self.old_django_email
+
+    def test_valid_activation(self):
+        """
+        Activating a user within the permitted window makes the
+        account active, and resets the activation key.
+
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+        activated = (self.registration_profile.objects
+                     .activate_user(profile.activation_key))
+
+        self.failUnless(isinstance(activated, UserModel()))
+        self.assertEqual(activated.id, new_user.id)
+        self.failIf(activated.is_active)
+
+        profile = self.registration_profile.objects.get(user=new_user)
+        self.assertTrue(profile.activated)
+
+    def test_valid_activation_with_profile(self):
+        """
+        Activating a user within the permitted window makes the
+        account active, and resets the activation key.
+
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+        activated_profile = (self.registration_profile.objects
+                             .activate_user(profile.activation_key, get_profile=True))
+
+        self.failUnless(isinstance(activated_profile, self.registration_profile))
+        self.assertEqual(activated_profile.id, profile.id)
+        self.failUnless(activated_profile.activated)
+
+        new_user.refresh_from_db()
+        self.assertTrue(activated_profile.user.id, new_user.id)
+        self.assertFalse(new_user.is_active)
+
+    def test_resend_activation_email_activated_user(self):
```

Comment:
Same as two functions above. Also, when using the admin_approve backend after activation outbox will not be empty, instead it will have one email, the mail sent to admins.